hastory-server/test/Data/Hastory/ServerSpec.hs

```diff
@@ -28,10 +30,25 @@ spec =
           Left (FailureResponse _ resp) <- runClientM req siClientEnv
           responseBody resp `shouldBe` "Invalid Token provided."
           responseStatusCode resp `shouldBe` status403
-    context "with correct token" $
+    context "with correct token" $ do
       it "saves entry to database" $ \ServerInfo {..} ->
         forAllValid $ \syncRequest -> do
           _ <- runClientM (appendCommand siToken syncRequest) siClientEnv
           let selectEntries = fmap SQL.entityVal <$> SQL.selectList [] []
           dbEntries <- SQL.runSqlPool selectEntries siPool
-          dbEntries `shouldBe` [SyncRequest.toServerEntry syncRequest]
+          dbEntries `shouldBe` [toServerEntry syncRequest]
+      context "when same entry is sync'd twice" $ do
+        it "the db does not change between the first sync and the second sync" $ \ServerInfo {..} ->
+          forAllValid $ \syncRequest -> do
+            _ <- runClientM (appendCommand siToken syncRequest) siClientEnv
+            let selectEntries = SQL.selectList [] []
```

Comment:
Note that if you don't impose some ordering, the database could choose an ordering and return the results in a different order next time. That would make this test flaky and give a false-positive.
It won't happen with sqlite though, so we're fine.

Reply:
Added `Asc ServerEntryId`.